### DISCUSSION TITLE
How to support ClownWallet

### DISCUSSION THREAD

**Discussion by hugeEavan** [rainbow-me/rainbowkit#1925] [Q&A] [answered]
How to support ClownWallet

**Comment by 0xmkh** [comment]
Here is an example of [safeWallet](https://github.com/rainbow-me/rainbowkit/blob/main/packages/rainbowkit/src/wallets/walletConnectors/safeWallet/safeWallet.ts). You could follow the same code just tweak it and change it a bit based on your needs.

Or we could try adding it on our end if you want to. Do you know if Clown Wallet supports EVM chains ?

**Reply by hugeEavan** [reply, marked as answer]
Yes, it supports EVM

**Comment by NORAVemp** [comment]
@magiziz  I want to add our Clown wallet to your framework, what should I do?

**Reply by 0xmkh** [reply]
We have our docs example [here](https://www.rainbowkit.com/docs/custom-wallets#examples). Here is another example of [rabbyWallet](https://github.com/rainbow-me/rainbowkit/blob/main/packages/rainbowkit/src/wallets/walletConnectors/rabbyWallet/rabbyWallet.ts).

Let me know if this works 👍

**Comment by hugeEavan** [comment]
I would like to add our clown wallet to your terminal wallet list

**Reply by 0xmkh** [reply]
Sure. Is this the browser extension ? https://chromewebstore.google.com/detail/clown-wallet/bipdhagncpgaccgdbddmbpcabgjikfkn?hl=en

**Reply by hugeEavan** [reply]
yes

**Comment by hugeEavan** [comment]
hello,Can the plugin wallet clownwallet be integrated into your terminal wallet list?

**Reply by 0xmkh** [reply]
Hey we decided that it's best if you create your own wallet and add it to the RainbowKit's `getDefaultConfig`. Here is an example of how you can create your own custom wallet https://www.rainbowkit.com/docs/custom-wallets.
